Foundation/NSXMLParser.swift

```diff
@@ -73,48 +73,40 @@ internal func _NSXMLParserExternalEntityWithURL(interface: _CFXMLInterface, urlS
     
     switch policy {
     case .ResolveExternalEntitiesSameOriginOnly:
-        if let url = parser._url {
-            if a == nil {
-                a = NSURL(string: String(urlStr))
+        guard let url = parser._url else { break }
```

Comment:
This case looks reasonable to me the original code was a bit of a rat's nest but I think all edges are taken care of here.